ansible/roles/fanout/tasks/main.yml

```diff
@@ -6,36 +6,60 @@
 ### specified in this playbook, you would need to come up with your own fanout switch deployment 
 ### playbook
 ################################################################################################
-  # Gather minigraph facts
+# change the root fanout port vlans
+- set_fact:
+    dut: "{{ inventory_hostname }}"
+    connect_leaf: true
+    leaf: "{{ ansible_host }}"
+
+- include: rootfanout_connect.yml
+
+# Deploy leaf fanout switch
+- set_fact:
+    ansible_host: "{{ leaf }}"
+
 - name: Gathering lab graph facts about the device
   conn_graph_facts: host={{ inventory_hostname }}
   connection: local
-  tags: always
 
 - name: prepare fanout switch admin login info
   set_fact: ansible_ssh_user={{ fanout_admin_user }} ansible_ssh_pass={{ fanout_admin_password }}
 
 - block:
+  - name: create persistent shell login
+    shell: touch /mnt/flash/rc.eos
+
+  - name: save persistent shell login
+    template: src=rc.eos.j2
+              dest=/mnt/flash/rc.eos
+
   - name: build fanout startup config for Arista Fanout leaf
     template: src=arista_7260_deploy.j2
               dest=/mnt/flash/startup-config
-    notify:
-      - Restart the box
+  - name: reboot
+    shell: sleep 2 && shutdown -r now "Reboot"
+    async: 1
+    poll: 0
+    ignore_errors: true
   when: device_info.HwSku == "Arista-7260QX-64" 
+  become: yes
 
+- block:
+  - name: create persistent shell login
+    shell: touch /mnt/flash/rc.eos
 
-- name: build fanout startup config
-  action: apswitch template=force10_s6100_deploy.j2
-  connection: switch
-  args:
-    login: "{{ switch_login['Force10'] }}"
-  when: device_info.HwSku == "Force10-S6100"
+  - name: save persistent shell login
+    template: src=rc.eos.j2
+              dest=/mnt/flash/rc.eos
 
-- block:
   - name: build fanout startup config
     template: src=arista_7060_deploy.j2
               dest=/mnt/flash/startup-config
-    notify:
-       - Restart the box
-  when: device_info.HwSku == "Arista-7060CX-32S"
 
+  - name: reboot
+    shell: sleep 2 && shutdown -r now "Reboot"
+    async: 1
+    poll: 0
+    ignore_errors: true
+  when: device_info.HwSku == "Arista-7060CX-32S"
```

Comment:
can we unify them into one block?,

only the deploy deploy script is different.